### PR TITLE
Fix/navbar users admin only

### LINKED ISSUE #35: navbar Users item should only be visible to admin role
The intention is to allow (as a nice to have) `member`s to see profiles of article contributors. Only `admin`s should be able to see a list of all users and amend them from the Users list page.

- [x] implement roles in FE

### PR DESCRIPTION
**Pull request by nikrb**
display users navbar menu item only for admin users
pass user role to FE on login
add BE check for admin role auth for users list

fixes #35

### CODE REVIEW

File: backend/routes/auth/isLoggedInAdmin.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function isLoggedInAdmin(req, res, next) {
+  if (
+    process.env.NODE_ENV === 'test' ||
+    (req.isAuthenticated() && req.user.role === 'admin')
+  ) {
```
Comment (jenovs) [inline backend/routes/auth/isLoggedInAdmin.js]: I prefer doing something like this:
```js
const isTestEnv = process.env.NODE_ENV === 'test';
const isAuthenticated = req.isAuthenticated();
const hasAdminRole = req.user.role === 'admin';

if (isTestEnv || (isAuthenticated && hasAdminRole) {
...
}
```
It's a bit longer, but much more readable.

**Review by jenovs — APPROVED** [review]
Looks good, only BE tests are failing and FE tests warns about prop types :wink:

File: backend/routes/auth/isLoggedInAdmin.js (hunk 1 above)
Comment (nikrb) [reply backend/routes/auth/isLoggedInAdmin.js]: you have a point there, though I think the isAuthenticated is a bit OTT.

**Comment by nikrb** [comment]
thanx for catching the test failure/warnings

**Comment by jenovs** [comment]
BE tests are still failing

**Comment by nikrb** [comment]
@jenovs I'm seeing a clean test run front and back. Can you give me some detail as to what is failing for you?

**Comment by jenovs** [comment]
be test error

**Comment by nikrb** [comment]
hmmm I don't see that. can you try re-installing? I wasn't aware of any auth requirements in the test, but will have a look

**Comment by nikrb** [comment]
Not sure how to proceed here.
I have cloned the repo but can't reproduce the OAuth2 error.
Also looked through the backend tests, and we don't actually test the auth anywhere so not sure where the call to passport lib is coming from. Could you have something installed globally that's interfering somehow?
Open to suggestions.

**Comment by jenovs** [comment]
I think I found the problem: `backend/test/setup.js` line 4:
```js
require('dotenv').config({ path: '.env.test' });
```
README doesn't have anything about `.env.test`.

**Comment by nikrb** [comment]
oops - sorry about that dude, I, of course, have a .env.test. Thanks for fixing it, I'll update readme.

I've made the review changes but found a bug with the navbar for admin role, which kinda opened a can of worms which I need to fix, so will probably need another review when I'z done.

**Comment by nikrb** [comment]
had to remove hasAdminRole as req.user is not populated during test

**Pull request merged**